staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -512,6 +512,62 @@ func PollImmediateUntil(interval time.Duration, condition ConditionFunc, stopCh
 	}
 }
 
+// PollUntilWithContext tries a condition func until it returns true,
+// an error or the specified context is cancelled or expired.
+//
+// PollUntilWithContext always waits interval before the first run of 'condition'.
+// 'condition' will always be invoked at least once.
+func PollUntilWithContext(interval time.Duration, condition ConditionFunc, ctx context.Context) error {
```

Comment:
Actually - I think I'm against introducing this function.

Why can't just the caller do

```
PollUntil(interval, condition, ctx.Done())
```

?